pkg/telefork/client.go

```diff
@@ -0,0 +1,110 @@
+package telefork
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/getoutreach/gobox/pkg/log"
+)
+
+type Event map[string]interface{}
+
+type Client interface {
+	SendEvent(event Event)
+
+	AddField(key string, val interface{})
+	AddInfo(args ...log.Marshaler)
+
+	Close()
+}
+
+func NewClient(appName, apiKey string) Client {
+	endpoint := "https://telefork.outreach.io/"
+	if os.Getenv("OUTREACH_TELEFORK_ENDPOINT") != "" {
+		endpoint = os.Getenv("OUTREACH_TELEFORK_ENDPOINT")
+	}
+	return &client{
+		appName:  appName,
+		endpoint: endpoint,
+		apiKey:   apiKey,
+	}
+}
+
+type client struct {
+	appName  string
+	apiKey   string
+	endpoint string
+	events   []Event
+
+	commonProps map[string]interface{}
+}
+
+func (c *client) SendEvent(event Event) {
+	e := make(Event)
+
+	for k, v := range c.commonProps {
+		e[k] = v
+	}
+	for k, v := range event {
+		e[k] = v
+	}
+
+	// We don't want certain fields added by honeycomb automatically.
+	// We do add some of them under different keys we've used before in CLIs.
+	// We skip adding them for non-@outreach.io emails. (Since it can be used in OSS projects)
+	delete(e, "meta.beeline_version")
+	delete(e, "meta.local_hostname")
+	delete(e, "meta.span_type")
+
+	c.events = append(c.events, e)
+}
+
+func (c *client) Close() {
+	if c.apiKey == "" || c.apiKey == "NOTSET" {
+		return
```

Comment:
If things are not configured right then nothing will be sent and there is no error message, should we add a printlin that is a warning that this has occurred?

Reply:
I'm not sure this shouldn't just stay silent, telemetry is kinda optional (also, this is set during build, so if it's not, then that's build error not a runtime error)

Reply:
Ok sounds good.